core/src/main/java/lucee/runtime/tag/util/QueryParamConverter.java

```diff
@@ -231,14 +231,14 @@ private static SQLItems<SQLItem> get(String name, List<SQLItems<NamedSQLItem>> i
 		SQLItems<NamedSQLItem> item;
 		while (it.hasNext()) {
 			item = it.next();
-			if (item.isEmpty()) {
-				throw new ApplicationException("param [" + name + "] may not be empty");
-			}
-			if (item.get(0).name.equalsIgnoreCase(name)) {
+			if (item.size() > 0 && item.get(0).name.equalsIgnoreCase(name)) {
+				if (item.isEmpty()) {
+					throw new ApplicationException("queryparam [" + name + "] may not be empty");
+				}
 				return item.convertToSQLItems();
```

Comment:
old version first check if empty or not, if not empty it get the value of the first element.
new version check if empty after it already has checked the size>0, what is the same as not empty.
That makes no sense.